hypothesis/client
# Add visibility hidden to iframe when closed

- Add visibility hidden to the sidebar iframe when its closed (and after it finishes animation). This prevents all interaction from the components of the sidebar.

- Add an optional coerce method to the hostPageSetting so that any incoming values can be coerced if needed. This may be required for values from from via that may only be a string type.

- `openSidebar` config value gets coerced to a boolean in the annotator settings (just like it does in the host-config).

---------

Test in 2 ways:

1. change dev-server.js to have openSidebar false (or true) & restart the client server.
`is-closed` should be added when its `false` to the `<iframe>`  but not added otherwise.

2. Open and close the sidebar and the `is-closed` class should also be added or removed corresponding to the state of the sidebar after show/hide events are fired.


Fixes https://github.com/hypothesis/client/issues/2058
Fixes https://github.com/hypothesis/client/issues/2083
